datalad/downloaders/tests/test_providers.py

```diff
@@ -146,3 +151,78 @@ def test_Providers_from_config__files(sysdir, userdir, dsdir):
             providers = Providers.from_config_files(reload=True)
             provider = providers.get_provider('https://crcns.org/data....')
             assert_equal(provider.name, 'usercrcns')
+
+
+@with_tempfile(mkdir=True)
+def test_providers_enter_new(path):
+    with patch.multiple("appdirs.AppDirs", site_config_dir=None,
+                        user_config_dir=path):
+        providers_dir = op.join(path, "providers")
+        providers = Providers.from_config_files(reload=True)
+
+        url = "blah://thing"
+        url_re = "blah:\/\/.*"
+        auth_type = "http_auth"
+        creds = "user_password"
+
+        @with_testsui(responses=["foo", url_re, auth_type,
+                                 creds, "no"])
+        def no_save():
+            providers.enter_new(url)
+        no_save()
+        assert_false(op.exists(op.join(providers_dir, "foo.cfg")))
+
+        @with_testsui(responses=["foo", url_re, auth_type,
+                                 creds, "yes"])
+        def save():
+            providers.enter_new(url)
+        save()
+        ok_exists(op.join(providers_dir, "foo.cfg"))
+
+        create_tree(path=providers_dir, tree={"exists.cfg": ""})
+        @with_testsui(responses=["exists", "foobert", url_re,
+                                 auth_type, creds, "yes"])
+        def already_exists():
+            providers.enter_new(url)
+        already_exists()
+        ok_exists(op.join(providers_dir, "foobert.cfg"))
+
+        @with_testsui(responses=["crawdad", "yes"])
+        def known_provider():
+            providers.enter_new(url)
+        known_provider()
+
+        @with_testsui(responses=["foo2", url_re, auth_type,
+                                 creds, "yes"])
+        def auth_types():
+            providers.enter_new(url, auth_types=["http_basic_auth"])
+        auth_types()
+        ok_exists(op.join(providers_dir, "foo2.cfg"))
+
+        @with_testsui(responses=["foo3", "doesn't match", url_re, auth_type,
+                                 creds, "yes"])
+        def nonmatching_url():
+            providers.enter_new(url, auth_types=["http_basic_auth"])
+        nonmatching_url()
+        ok_exists(op.join(providers_dir, "foo3.cfg"))
```

Comment:
Lovely!
I guess the next one would be a similar one for `_enter_credentials`?

Reply:
I was hoping I had already fulfilled your request to help with your PR.

Reply:
you did help! ok,I could continue whenever I get a chance, but not sure when it would be :-(

Reply:
I wasn't saying I wouldn't help more, just stating what I had hoped :)

I'll look into filling out the tests more, starting with `_enter_credentials`.

Reply:
thanks in advance! ;-)